src/edata.cpp

```diff
@@ -404,12 +408,16 @@ void ConditionContext::restore()
     model->setParameterList(originalParameterList);
     // parameter scale has to be set before initial state sensitivities
     model->setParameterScale(originalScaling);
-    model->setInitialStates(originalx0);
-    model->setUnscaledInitialStateSensitivities(originalsx0);
+
+    if(!originalx0.empty())
+        model->setInitialStates(originalx0);
```

Comment:
Do we really need to check whether `originalx0` is empty here? I think we do want to reset Initial states if they were empty originally.

Reply:
I think if they were empty originally, them Model::x0 will still be empty after simulation. I think fx0 is not stored there if not manually set (in which case originalx0 here would be non-empty). With the current implementation it is not strictly required (we would just replace an empty vector by another empty vector), but I think it doesn't hurt. Maybe I am overlooking something.

Reply:
I think you are right, fine as is.